modules/Emails/EmailsController.php

```diff
@@ -154,6 +161,116 @@ public function action_send()
         $this->view = 'sendemail';
     }
 
+    /**
+     * Parse and replace bean variables
+     * but first validate request,
+     * see log to check validation problems
+     *
+     * return Email bean
+     *
+     * @param Email $email
+     * @param array $request
+     * @return Email
+     */
+    protected function replaceEmailVariables(Email $email, $request) {
+
+        // request validation before replace bean variables
+
+        if ($this->isValidRequestForReplaceEmailVariables($request)) {
+
+            $macro_nv = array();
+
+            $focusName = $request['parent_type'];
+            $focus = BeanFactory::getBean($focusName, $request['parent_id']);
+            if ($email->module_dir == 'Accounts') {
+                $focusName = 'Accounts';
+            }
+
+            /**
+             * @var EmailTemplate $emailTemplate
+             */
+            $emailTemplate = BeanFactory::getBean(
+                'EmailTemplates',
+                isset($request['emails_email_templates_idb']) ?
+                    $request['emails_email_templates_idb'] :
+                    null
+            );
+            $templateData = $emailTemplate->parse_email_template(
+                array(
+                    'subject' => $email->name,
+                    'body_html' => $email->description_html,
+                    'body' => $email->description,
+                ),
+                $focusName,
+                $focus,
+                $macro_nv
+            );
+
+            $email->description_html = $templateData['body_html'];
+            $email->description = $templateData['body'];
+
+        } else {
+
+            $this->log('Email variables is not replaced because an invalid request.');
+
+        }
+
+
+        return $email;
+    }
+
+    /**
+     * Request validation before replace bean variables,
+     * see log to check validation problems
+     *
+     * @param array $request
+     * @return bool
+     */
+    protected function isValidRequestForReplaceEmailVariables($request) {
+
+        $ok = true;
```

Comment:
OK can === false, which is a little confusing. Can we perhaps change this to $response?